### PR TITLE
Improved CS

### PR DESCRIPTION
**Pull request by nomack84**
Improved CS and applied ES6 whenever possible, changing `var` by `let` and also `for` by `for..of`

### CODE REVIEW

File: src/headers.js [hunk 1]
```diff
@@ -30,21 +30,20 @@ export class Headers {
...
+    let headers = new Headers();
     if (!headerStr) {
       return headers;
     }
 
-    var headerPairs = headerStr.split('\u000d\u000a');
-    for (var i = 0; i < headerPairs.length; i++) {
-      var headerPair = headerPairs[i];
+    let headerPairs = headerStr.split('\u000d\u000a');
+    for (let headerPair of headerPairs) {
```
Comment (bryanrsmith) [inline src/headers.js]: I certainly agree that for..of is nicer here, but we've intentionally avoided it because babel generates a lot of code to handle iterators, which bloats the lib.

File: src/http-response-message.js [hunk 2]
```diff
@@ -3,60 +3,68 @@ import {Headers} from './headers';
...
-  constructor(requestMessage : RequestMessage, xhr : XHR, responseType : string, reviver : Function){
-    this.requestMessage = requestMessage;
-    this.statusCode = xhr.status;
-    this.response = xhr.response || xhr.responseText;
-    this.isSuccess = xhr.status >= 200 && xhr.status < 400;
-    this.statusText = xhr.statusText;
-    this.reviver = reviver;
-    this.mimeType = null;
+  constructor(requestMessage : RequestMessage, xhr : XHR, responseType : string, reviver : Function) {
+    let me = this;
```
Comment (bryanrsmith) [inline src/http-response-message.js]: What's the reason for this change? I think it's a bit harder to read.

File: src/jsonp-request-message.js [hunk 3]
```diff
@@ -25,22 +25,22 @@ class JSONPXHR {
...
-    script.onerror = (e) => {
+    script.onerror = function(e) {
       cleanUp();
 
       this.status = 0;
       this.onerror(new Error('error'))
     };
 
-    let cleanUp = () => {
+    let cleanUp = function() {
```
Comment (bryanrsmith) [inline src/jsonp-request-message.js]: This probably broke the function because we've lost the binding to 'this'.

**Comment by EisenbergEffect** [comment]
Using `let` is fine but we don't want to use `for..of` at this time because it not only generates more code but the code is much less performance that the original loop and that performance is really important for us at the framework level.

File: src/http-response-message.js (hunk 2 above)
Comment (nomack84) [inline src/http-response-message.js]: I've seen this technique in other frameworks, ExtJS for instance, it helps the uglifier to minify the code. The gain is not big, perhaps it doesn't justify it.

**Comment by nomack84** [comment]
Understood, I'll remove the `for..of`

File: src/http-response-message.js (hunk 2 above)
Comment (bryanrsmith) [inline src/http-response-message.js]: I'd prefer to keep `this`.

**Comment by bryanrsmith** [comment]
@nomack84 Do you think you'll be able to address the outstanding issues noted in the comments? It'd also be great if you could squash the extra commits and update the commit message to follow our contributing guidelines.

**Comment by nomack84** [comment]
Yeah, I do. :+1:

**Comment by nomack84** [comment]
@bryanrsmith I've fixed all issues and updates the commit message. I check other commits and I've found many with a commit like this, so I guess is fine this way. If you think is not, please tell me what to put to update it quickly and merge this.

**Comment by EisenbergEffect** [comment]
@bryanrsmith Please review. CLA is confirmed.

**Comment by EisenbergEffect** [comment]
Closing this for now. There were conflicts with the last repo maintenance pass and I didn't realize it ;(

**Comment by EisenbergEffect** [comment]
Have a look at the master and see if there's anything else that needs cleanup.
